src/vm/arm64/asmhelpers.S

```diff
@@ -310,29 +310,46 @@ LOCAL_LABEL(shadowupdateend):
     ldp  x12, x13, [sp],#16
 #endif
 
-    // Branch to Exit if the reference is not in the Gen0 heap
+    // Branch to DoneCardTable if the reference is not in the Gen0 heap
```

Comment:
is there a reason why you chose to do this in the reversed order as we normally do? in the existing barriers we always update the WW table first, then jump to the CheckCardTable* labels. it would be good to keep that order.